lib/generator.rb

```diff
@@ -35,13 +37,43 @@ def test_cases
     cases.call(data)
   end
 
+  def metadata_repository_missing_message
+    <<-EOM.gsub(/^ {6}/, '')
+
+      '#{METADATA_REPOSITORY}' repository not found.
+      Try running the command:
+        git clone https://github.com/exercism/#{METADATA_REPOSITORY}.git "#{metadata_dir}"
```

Comment:
The two stated concerns I was hearing was "short message" and "copy and paste".  The coding change addresses both values.  If they are not worth the code, then this is likely good enough, and that was the intended (though apparently not well communicated) message.

Here is what it may look like though, if the two expressed values are worth the effort...

```ruby
  def metadata_repository_missing_message
    <<-EOM.gsub(/^ {6}/, '')

      '#{METADATA_REPOSITORY}' repository not found.
      Try running the command:
        git clone #{preferred_protocol}#{METADATA_REPOSITORY}.git "#{metadata_dir}"

    EOM
  end

  def preferred_protocol
    { true => 'git@github.com:exercism/',
      false => 'https://github.com/exercism/'
    }[IO.read('.git/config').include?('git@github')] 
# The above line is sloppy, as it does not check for 
# the precise scope of 'origin', and as 'origin' is a 
# common, though arbitrary, name and does not 
# cover 100%.
  end
```

There still may be some advantage to determine set "PROJECT_ROOT" so that it can be used for both the .git/ and the ../x-common folders.

If this is incorporated, then more people can simply copy and paste, which I think is worth a little bit of extra oomph.

If it isn't, then just consider this my "Looks good to me" vote.